log4j-core/src/main/java/org/apache/logging/log4j/core/impl/ThreadContextDataInjector.java

```diff
@@ -62,21 +62,9 @@
     public static Collection<ContextDataProvider> contextDataProviders =
             new ConcurrentLinkedDeque<>();
 
-    private static volatile List<ContextDataProvider> serviceProviders = null;
-    private static final Lock providerLock = new ReentrantLock();
+    private static final List<ContextDataProvider> serviceProviders = getServiceProviders();
```

Comment:
Our static finals (constants) should be in upper-case.

Reply:
Good call, I'll update the initializer such that the list itself is unmodifiable as well.